zk/stages/stage_sequence_execute_utils.go

```diff
@@ -204,6 +204,11 @@ func prepareForkId(lastBatch, executionAt uint64, hermezDb forkDb) (uint64, erro
 		if err := hermezDb.WriteForkIdBlockOnce(latest, executionAt+1); err != nil {
 			return latest, err
 		}
+		// write the fork id for this batch now so that the remote executor receives
+		// the correct fork id for the batch
+		if err = hermezDb.WriteForkId(nextBatch, latest); err != nil {
```

Comment:
Why we're writing the forkId here when it will be written in the "finalizeBlock" which is again before sending the request to an executor?

Reply:
Without it we get this 
<img width="1064" alt="image" src="https://github.com/user-attachments/assets/a149fb43-e24c-4815-ba9d-af91c449cc94">